src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs

```diff
@@ -30,19 +32,39 @@ public sealed class UnmanagedPSEntry
         /// <param name="argc">
         /// Length of the passed in argument array.
         /// </param>
+        [Obsolete("Do not use the method", error: true)]
         public static int Start(string consoleFilePath, [MarshalAs(UnmanagedType.LPArray, ArraySubType = UnmanagedType.LPWStr, SizeParamIndex = 2)]string[] args, int argc)
         {
+            return Start(args);
+        }
+
+        /// <summary>
+        /// Starts managed MSH.
+        /// </summary>
+        /// <param name="args">
+        /// Command line arguments to the managed MSH
+        /// </param>
+        public static int Start(string[] args)
+        {
+            if (args == null)
+            {
+                throw new ArgumentNullException(nameof(args));
+            }
+
             // Warm up some components concurrently on background threads.
             EarlyStartup.Init();
 
-            // We need to read the settings file before we create the console host
-            CommandLineParameterParser.EarlyParse(args);
+            var banner = string.Format(
+                CultureInfo.InvariantCulture,
+                ManagedEntranceStrings.ShellBannerNonWindowsPowerShell,
+                PSVersionInfo.GitCommitId);
+
+            ConsoleHost.ParseCommandLine(args);
```

Comment:
I don't know what this change really does, but it seems like a bigger change than the others. @daxian-dbw should have a look here

Reply:
> I don't know 

I put full change list in the PR description. Can I add more for you?